src/XMakeBuildEngine/Evaluation/Evaluator.cs

```diff
@@ -2086,10 +2079,11 @@ private List<ProjectRootElement> ExpandAndLoadImports(string directoryOfImportin
             //
 
             // Adding the value of $(MSBuildExtensionsPath*) property to the list of search paths
-            var prop = _data.GetProperty(refKindInProject.StringRepresentation);
+            var prop = _data.GetProperty(fallbackSearchPathMatch.PropertyName);
+            var fallbackExtensionPaths = fallbackSearchPathMatch.SearchPaths.ToList();
             fallbackExtensionPaths.Insert(0, prop.EvaluatedValue);
```

Comment:
I'd like to see this block more explicit, and inserting at item 0 of a list reads wrong to me.

My instinct is something like
```C#
var pathsToSearch =
    // The actual value of the property, with no fallbacks
    new [] { prop.EvaluatedValue}
    // The list of fallbacks, in order
    .Concat(fallbackSearchPathMatch.SearchPaths).ToList();
```

Though I don't love that either.